internal/user/user.go

```diff
@@ -0,0 +1,23 @@
+package user
+
+import "github.com/clean-code-projects/co-coders-api/internal/criteria"
+
+// User ..
+type User struct {
+	CollabStyles []criteria.CollabStyle
+}
+
+// HasCollabStyle ..
+func (u User) HasCollabStyle(collabStyle criteria.CollabStyle) bool {
+	for _, style := range u.CollabStyles {
+		if style == collabStyle {
+			return true
+		}
+	}
+	return false
+}
+
+// New ..
+func New(collabStyles []criteria.CollabStyle) User {
```

Comment:
Maybe this is a go thing.
Why use user.New instead of just User and setting the object directly?
Is it the difference between a struct and a type?

Reply:
Using a New function as a constructor is a go convention. We don't need to follow it, but it has its benefits